ecosystem/sep-0006.md

```diff
@@ -589,29 +793,33 @@ The response should be a JSON object like:
+* `fee_percent`: Optional percentage fee for deposit, in percentage points of the deposited **off-chain** asset. Leave blank if there is no fee or the fee schedule is complex.
 * `fields` object as explained below.
 
+_Note: the attributes above mentioned are valid for both kinds of deposits `deposit` and `deposit-exchange`, exept for the `fee_*` attributes that are only valid for `deposit`._
+
 #### For each withdrawal asset, response contains:
 
 * `enabled`: `true` if SEP-6 withdrawal for this asset is supported
 * `authentication_required`: Optional. `true` if client must be [authenticated](#authentication) before accessing the deposit endpoint for this asset. `false` if not specified.
 * `min_amount`: Optional minimum amount. No limit if not specified.
 * `max_amount`: Optional maximum amount. No limit if not specified.
-* `fee_fixed`: Optional fixed (flat) fee for withdraw. In units of the deposited asset. Leave blank if there is no fee or the fee schedule is complex.
-* `fee_percent`: Optional percentage fee for withdraw. In percentage points. Leave blank if there is no fee or the fee schedule is complex.
+* `fee_fixed`: Optional fixed (flat) fee for withdraw, in units of the Stellar asset. Leave blank if there is no fee or the fee schedule is complex.
+* `fee_percent`: Optional percentage fee for withdraw, in percentage points of the Stellar asset. Leave blank if there is no fee or the fee schedule is complex.
 * a `types` field with each type of withdrawal supported for that asset as a key. Each type can specify a `fields` object as below explaining what fields are needed and what they do.
 
-The `fields` object allows an anchor to describe fields that are passed into `/deposit` and `/withdraw`. It can explain standard fields like `dest` and `dest_extra` for withdrawal, and it can also specify extra fields that should be passed into `/deposit` or `/withdraw` such as an email address or bank name. If a field is part of the KYC/AML flow handled by SEP-12 or the field is handled by your interactive deposit/withdrawal flow, there's no need to list it in `/info`. Only fields that are passed to `/deposit` or `/withdraw` need appear here.
+_Note: the attributes above mentioned are valid for both kinds of withdrawals `withdraw` and `withdraw-exchange`, exept for the `fee_*` attributes that are only valid for `withdraw`._
+
+The `fields` object allows an anchor to describe fields that are passed into `/deposit*` and `/withdraw*`. It can explain standard fields like `dest` and `dest_extra` for withdrawal, and it can also specify extra fields that should be passed into `/deposit*` or `/withdraw*` such as an email address or bank name. If a field is part of the KYC/AML flow handled by SEP-12 or the field is handled by your interactive deposit/withdrawal flow, there's no need to list it in `/info`. Only fields that are passed to `/deposit*` or `/withdraw*` need appear here.
```

Comment:
❓ What if we need to add new fields to exchange and not the other, will they be included in the shared list? Could this get confusing?

Reply:
Sorry, I didn't understand your point here.

Reply:
The section notes that the `fields` object describes fields that are passed into all four endpoints (/deposit, /deposit-exchange, etc). What happens if we want to add a field only to the exchange version?

Reply:
Good point! How does this look? https://github.com/stellar/stellar-protocol/pull/973/commits/04234d98cff2d4a13615ea10ddb49aee45dbb18e

Reply:
04234d9 looks like it changes fields that are not part of the `fields` object.  I think my comment was misunderstood. I was referring to this section about the `fields` object:
>The `fields` object allows an anchor to describe fields that are passed into `/deposit*` and `/withdraw*`. It can explain standard fields like `dest` and `dest_extra` for withdrawal, and it can also specify extra fields that should be passed into `/deposit*` or `/withdraw*` such as an email address or bank name.

Reply:
Ok got it, Thanks for explaining again.

Yes, the idea here is `fields` can contain different values for any of the following: `/deposit`, `/withdraw`, `/deposit-exchange` and `/withdraw-exchange`.

I thought using the `*` in `/deposit*` to represent both `/deposit` and `/deposit-exchange` would be enough to make it clear... what do you think of this copy https://github.com/stellar/stellar-protocol/pull/973/commits/d5d591481e4802587229bd6749b5c40202ceef06?

Reply:
That does read clearer by being verbose, but that wasn't actually what I had an issue with. My question was, "What happens if we want to add a field only to the exchange version?"

If we want to add a field X that is only to be used with "/deposit-exchange" and "/withdraw-exchange", and if I add field X to `fields`, will the field also automatically apply to "/deposit" and "/withdraw"?

I ask this because we went with separate endpoints to remove optionality and overlap, to separate them, but in this `fields` definition we are joining them back together.

Reply:
If you look at the `GET \info` [response](https://github.com/stellar/stellar-protocol/blob/04234d98cff2d4a13615ea10ddb49aee45dbb18e/ecosystem/sep-0006.md#response-4) you'll notice there is an occurrence of `fields` for every one of those endpoints:

```js
{
  "deposit": {
    "USD": {
      "fields": {...}
    },
  },
  "deposit-exchange": {
    "USD": {
      "authentication_required": true,
      "fields": {...}
    }
  },
  "withdraw": {
    "USD": {
      "types": {
        "bank_account": {
          "fields": {...}
        }
      }
    },
  },
  "withdraw-exchange": {
    "USD": {
      "types": {
        "bank_account": {
          "fields": {...}
        }
      }
    }
  },
}
```

That should be enough to allow every one of those endpoints to have different `fields`.

Reply:
Oh right, I misunderstood how `fields` works. 🎉 Thanks for explaining.